signal/signal.go

```diff
@@ -45,6 +48,10 @@ func Intercept() {
 // callback registration.
 // It must be run as a goroutine.
 func mainInterruptHandler() {
+	// Close our started channel to signal that we have started listening
+	// for interrupts.
+	close(started)
```

Comment:
This would cause a double close panic if `Intercept` is ever called twice....

Perhaps we should try to enforce that there is only a singleton registration for interrupt signals?